aws/resource_aws_neptune_parameter_group_test.go

```diff
@@ -0,0 +1,218 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/neptune"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSNeptuneParameterGroup_basic(t *testing.T) {
+	var v neptune.DBParameterGroup
+	rName := fmt.Sprintf("parameter-group-test-terraform-%d", acctest.RandInt())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSNeptuneParameterGroupDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccAWSNeptuneParameterGroupConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSNeptuneParameterGroupExists("aws_neptune_parameter_group.bar", &v),
+					testAccCheckAWSNeptuneParameterGroupAttributes(&v, rName),
+					resource.TestCheckResourceAttr(
+						"aws_neptune_parameter_group.bar", "name", rName),
+					resource.TestCheckResourceAttr(
+						"aws_neptune_parameter_group.bar", "family", "neptune1"),
+					resource.TestCheckResourceAttr(
+						"aws_neptune_parameter_group.bar", "description", "Managed by Terraform"),
+					resource.TestCheckResourceAttr(
```

Comment:
Minor nitpick: when testing nested attributes like TypeList and TypeSet, we should check the associated value count as well: `resource.TestCheckResourceAttr("aws_neptune_parameter_group.bar", "parameter.#", "1")`

We should apply this to any of the other tests below where we also expect a certain number of parameters (especially the removal test).